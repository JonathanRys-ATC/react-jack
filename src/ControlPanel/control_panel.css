.ControlPanel_main {
    min-width: 220px;
    width: 20%;
    right: 0;
    margin: 15px;
    padding: 25px;
    padding-top: 20px;
    position: absolute;
    border: 1px solid black;
    border-radius: 10px;
}

.ControlPanel_main_panel-header {
    font-size: 1.5em;
    font-weight: bold;
}