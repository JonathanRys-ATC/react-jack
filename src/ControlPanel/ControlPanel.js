import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './control_panel.css'

import { Button, Form, Label } from 'reactstrap'

export default class ControlPanel extends Component {
    componentWillReceiveProps(nextProps) {
        if (nextProps.dealer.hand.length < 1 || nextProps.player.hands[0].length < 2) {
            nextProps.keepDealing()
        } else if (!nextProps.turn.playersTurn) {
            //is it the dealer's turn?
            console.log("Not the player's turn.")
            console.log("stands?", nextProps.player.playerStands)
            console.log("busted?", nextProps.player.busted)
            console.log("hasBlackJack?", nextProps.player.hasBlackJack)
            if (nextProps.player.playerStands || nextProps.player.busted || nextProps.player.hasBlackJack) {
                //Take dealer's turn
                console.log("Looks like the dealer's turn:")

            }
            else {
                nextProps.nextPlayer()
            }
        }
    }

    render() {
        const canSplit = this.props.player.hands[0] && this.props.player.hands[0].length === 2 && this.props.player.hands[0][0].slice(1) === this.props.player.hands[0][1].slice(1)
        const canBuyInsurance = () => {
            if (!this.props.turn.isPlaying) return false
            if (!this.props.dealer.hand[0]) return false
            if (this.props.dealer.hand[0].slice(1) !== "A") return false
            if (!this.props.player.hasInsurance[this.props.player.handIndex]) return false
            return true;
        }

        return (
            <Form xs="4" sm="6" md="8" lg="12" className="ControlPanel_main">
                <Label className="ControlPanel_main_panel-header">Control Panel</Label>
                <Button size="md" color="secondary" block disabled={!this.props.turn.isPlaying} onClick={this.props.hitOnClick}>Hit</Button>
                <Button size="md" color="secondary" block disabled={!this.props.turn.isPlaying} onClick={this.props.standOnClick}>Stand</Button>
                <Button size="md" color="secondary" block disabled={!canSplit} onClick={this.props.splitOnClick}>Split</Button>
                <Button size="md" color="secondary" block disabled={this.props.player.hands[0].length !== 2} onClick={this.props.doubleDownOnClick}>Double Down</Button>
                <Button size="md" color="secondary" block disabled={!canBuyInsurance()} onClick={this.props.buyInsuranceOnClick}>Buy Insurance</Button>
                <Button size="md" color="primary" block disabled={this.props.turn.isPlaying} onClick={this.props.dealOnClick}>Deal</Button>
            </Form>
        )
    }
}
// Write deep prop type checks here
ControlPanel.propTypes = {
    drawnCard: PropTypes.string,
    isPlaying: PropTypes.bool,
}
