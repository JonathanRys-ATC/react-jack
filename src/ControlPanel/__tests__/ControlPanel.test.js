import React from 'react'
import { shallow } from 'enzyme'

import ControlPanel from "../ControlPanel"

import mockStore from "../../mockStore"

const mockProps = {
    ...mockStore,
    hitOnClick: jest.fn(),
    standOnClick: jest.fn(),
    winBet: jest.fn(),
    loseBet: jest.fn(),
    splitOnClick: jest.fn(),
    doubleDownOnClick: jest.fn(),
    buyInsuranceOnClick: jest.fn(),
    dealOnClick: jest.fn(),
    reset: jest.fn(),
}

describe("Test PlayField component", () => {
    it("matches the snapshot", () => {
        const testProps = { ...mockProps }
        const component = shallow(<ControlPanel {...testProps} />)
        expect(component).toMatchSnapshot()
    })
})
