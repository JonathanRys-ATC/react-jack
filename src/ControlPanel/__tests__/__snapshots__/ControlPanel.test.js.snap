// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlayField component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlPanel
    buyInsuranceOnClick={[MockFunction]}
    dealOnClick={[MockFunction]}
    dealer={
      Object {
        "avatar": "./images/dealer.png",
        "busted": Array [
          false,
        ],
        "hand": Array [
          "HA",
          "C5",
        ],
        "handFlipped": false,
        "hasBlackjack": Array [
          false,
        ],
        "name": "Dealer",
        "score": Array [
          0,
        ],
      }
    }
    deck={
      Object {
        "deck": Array [],
        "drawnCard": null,
      }
    }
    doubleDownOnClick={[MockFunction]}
    hitOnClick={[MockFunction]}
    loseBet={[MockFunction]}
    player={
      Object {
        "avatar": "./images/default.png",
        "balance": 0,
        "busted": Array [
          false,
        ],
        "currentBet": 0,
        "hands": Array [
          Array [
            "SA",
            "CK",
          ],
          Array [
            "D2",
            "HJ",
          ],
        ],
        "hasBlackjack": Array [
          false,
        ],
        "hasInsurance": Array [
          false,
        ],
        "name": "Player1",
        "playerIndex": 1,
        "playerStands": Array [
          false,
        ],
        "score": Array [
          0,
        ],
        "splitHand": false,
      }
    }
    reset={[MockFunction]}
    splitOnClick={[MockFunction]}
    standOnClick={[MockFunction]}
    turn={
      Object {
        "errMsg": "",
        "error": false,
        "isPlaying": true,
        "numPlayers": 2,
        "playersTurn": 0,
      }
    }
    winBet={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Label
          className="ControlPanel_main_panel-header"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Control Panel
        </Label>,
        <Button
          block={true}
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-point-down"
          />
          Hit
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-paper"
          />
          Stand
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-scissors"
          />
          Split
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          <FontAwesome
            name="hand-peace"
          />
          Double Down
        </Button>,
        <Button
          block={true}
          color="secondary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          Buy Insurance
        </Button>,
        <Button
          block={true}
          color="primary"
          disabled={true}
          onClick={[MockFunction]}
          size="md"
          tag="button"
        >
          Deal
        </Button>,
      ],
      "className": "ControlPanel_main",
      "lg": "12",
      "md": "8",
      "sm": "6",
      "tag": "form",
      "xs": "4",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Control Panel",
          "className": "ControlPanel_main_panel-header",
          "tag": "label",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": "Control Panel",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-point-down"
            />,
            "Hit",
          ],
          "color": "secondary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-point-down",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Hit",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-paper"
            />,
            "Stand",
          ],
          "color": "secondary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-paper",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Stand",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-scissors"
            />,
            "Split",
          ],
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-scissors",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Split",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": Array [
            <FontAwesome
              name="hand-peace"
            />,
            "Double Down",
          ],
          "color": "secondary",
          "disabled": false,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "hand-peace",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "Double Down",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": "Buy Insurance",
          "color": "secondary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": "Buy Insurance",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "block": true,
          "children": "Deal",
          "color": "primary",
          "disabled": true,
          "onClick": [MockFunction],
          "size": "md",
          "tag": "button",
        },
        "ref": null,
        "rendered": "Deal",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Label
            className="ControlPanel_main_panel-header"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Control Panel
          </Label>,
          <Button
            block={true}
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-point-down"
            />
            Hit
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-paper"
            />
            Stand
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-scissors"
            />
            Split
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            <FontAwesome
              name="hand-peace"
            />
            Double Down
          </Button>,
          <Button
            block={true}
            color="secondary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            Buy Insurance
          </Button>,
          <Button
            block={true}
            color="primary"
            disabled={true}
            onClick={[MockFunction]}
            size="md"
            tag="button"
          >
            Deal
          </Button>,
        ],
        "className": "ControlPanel_main",
        "lg": "12",
        "md": "8",
        "sm": "6",
        "tag": "form",
        "xs": "4",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Control Panel",
            "className": "ControlPanel_main_panel-header",
            "tag": "label",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": "Control Panel",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-point-down"
              />,
              "Hit",
            ],
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-point-down",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Hit",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-paper"
              />,
              "Stand",
            ],
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-paper",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Stand",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-scissors"
              />,
              "Split",
            ],
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-scissors",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Split",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": Array [
              <FontAwesome
                name="hand-peace"
              />,
              "Double Down",
            ],
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "hand-peace",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Double Down",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": "Buy Insurance",
            "color": "secondary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": "Buy Insurance",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "block": true,
            "children": "Deal",
            "color": "primary",
            "disabled": true,
            "onClick": [MockFunction],
            "size": "md",
            "tag": "button",
          },
          "ref": null,
          "rendered": "Deal",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
